File: python/display_APS_frames.py
###############################
# # plot frame images obtained
# # from net_raw_UDP_jAER
# # federico.corradi@inilabs.com
###############################

import pylab as plt
from time import sleep

def display_APS_frames(frames, time_interval=1):
    '''
    display DAVIS240 frames with fixed time interval
    inputs:
        frames : obtained from net_raw_UDP_jAER.py
        time_interval : optional parameter, time interval in between frames
    '''
    plt.ion()
    my_obj = plt.imshow(plt.rot90(frames[1][3, :, :]))
    plt.hold('on')
    for i in range(len(frames)):
        frames[i][3, :, :] = frames[i][2, :, :] - frames[i][1, :, :]
        frames[i][6, :, :] = frames[i][5, :, :] - frames[i][4, :, :]
        frames[i][3, :, :][frames[i][3, :, :] < 0] = 0
        my_obj.set_data(plt.rot90(frames[i][3, :, :]))
        plt.draw()
        sleep(1)

File: misc/davis/displayAPSframes.m
function displayAPSframes(frames, time_interval)
% diplay frames with a fixed time interval
%   inputs:
%       frames -> required. It is obtained from getAPSframesDavisGS
%       time_interval  -> optional param. time interval between frames
%
%   author federico.corradi@inilabs.com

    if nargin < 2
       time_interval = 0.1
    end

    [chan, x, y, numFrames] = size(frames);     %get infos on frames
    bw_cdsSignal = 3;                           %here are stored the bw values (resetbuffer - readbuffer)
    max_gray = 512;                             %constant?
    
    imshow(rot90(squeeze(frames(bw_cdsSignal,:,:,1))),[0,max_gray]);
    hold on
    for i=1:numFrames
        %check cdsSignal is non negative
        index_neg = frames(bw_cdsSignal,:,:,i) < 0;
        frames(bw_cdsSignal(index_neg),:,:,i) = 0;          %set cdsSignal = 0 if neg
        %all your other code here
        imshow(rot90(squeeze(frames(bw_cdsSignal,:,:,i))),[0,max_gray]);
        pause(time_interval);
    end

end
